src/language/records.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+_Records expressions_ are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+_Record type annotations_ are comma-delimited lists of types enclosed in parenthesis.
+The `(int, int)` statements here are record type annotations used as a return
+type and as parameter types:
+
+```dart
+(int, int) swap((int, int) record) {
+  var (a, b) = record;
+  return (b, a);
+}
+```
+
+Named fields inside _record type annotations_ go inside a brace-delimited section
+of type and name pairs, after all positional fields:
+
+```dart
+// Record type annotation in a variable declaration
+(String, String?, {int a, bool b}) record;
+```
+
+Naming fields in record type annotations is optional for documentation purposes.
+The names of those named fields are not optional, though,  
+once you create a record with named fields. The field names become part of the
+[record's type definition](#record-types). They can be omitted, but not replaced.
+
+For more information, see the [Record equality](#record-equality) section.
```

Comment:
**nit:** Prefer different construction because there is no call to action for the reader.
```suggestion
To learn more, check out the [Record equality](#record-equality) section.
```